examples/parallel_reduce/pi/common.h

```diff
@@ -0,0 +1,48 @@
+/*
+    Copyright (c) 2023 Intel Corporation
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#ifndef TBB_examples_pi_H
+#define TBB_examples_pi_H
+
+#include <cstdlib>
+
+typedef std::size_t number_t;
+typedef double pi_t;
+
+extern const number_t chunk_size;
+extern number_t num_intervals;
+extern pi_t inv_intervals;
+
+extern bool silent;
+
+inline pi_t pi_kernel(number_t i) {
+    pi_t dx = (pi_t(i) + pi_t(0.5)) * inv_intervals;
+    return pi_t(4.0) / (pi_t(1.0) + dx * dx);
```

Comment:
In terms of integration this code makes more sense (for me), when `inv_intervals` is called something like `step` because `inv_intervals` is not actually just inverted `num_intervals` but a result of a formula 
```math
step=\frac{b-a}{nintervals}
```
Where `a` and `b` represent the range of integration (from 0 to 1).

Reply:
Done